### DISCUSSION TITLE
The algorithm used to choose the map when switching rulesets could be improved.

### DISCUSSION THREAD

**Discussion by ziv-vy** [ppy/osu#29602] [Ideas]
### Type

Game behaviour

### Bug description

Swapping between rulesets _should_ constantly swap you between the same two beatmaps, but that's not the case!

### Screenshots or videos

https://youtu.be/ad-jpcadrNI

### Version

2024.817.0-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16741245/compressed-logs.zip)

**Comment by Joehuu** [comment]
Video is private.

**Comment by ziv-vy** [comment]
Fixed
